ids/format_test.go

```diff
@@ -0,0 +1,57 @@
+package ids
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestFormat(t *testing.T) {
+	type (
+		idInterface interface {
+			String() string
+			Hex() string
+		}
+		test struct {
+			id   idInterface
+			want map[string]string // format -> output
+		}
+	)
+	makeTestCase := func(id idInterface) test {
+		return test{
+			id: id,
+			want: map[string]string{
+				"%v":  id.String(),
+				"%s":  id.String(),
+				"%q":  `"` + id.String() + `"`,
+				"%x":  id.Hex(),
+				"%#x": `0x` + id.Hex(),
+			},
+		}
+	}
+
+	tests := []test{
+		makeTestCase(ID{}),
+		makeTestCase(GenerateTestID()),
+		makeTestCase(GenerateTestID()),
+		makeTestCase(ShortID{}),
+		makeTestCase(GenerateTestShortID()),
+		makeTestCase(GenerateTestShortID()),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.id.String(), func(t *testing.T) {
+			for format, want := range tt.want {
+				if got := fmt.Sprintf(format, tt.id); got != want {
```

Comment:
I'm curious to know the use case for this PR. We don't pass ID into Sprintf but use the logger, and there we use zap.Stringer.

Reply:
In an EVM implementation `ids.ID` is typically derived from a `common.Hash` (both `[32]byte`), which is idiomatically rendered as hex. My errors that include a block hash/ID use `%#x`.